[rebranch] Fix recently added merge function test

bnbarham opened the pull request:
This was just added in main, but rebranch has opaque pointers enabled all the time and there's already a `merge_func.ll` test.

bnbarham commented:
@swift-ci please test

aschwaighofer commented:
This test was added to test a specific failure (merging functions with casted return types).

Is there a copy of this test case in another file?

aschwaighofer commented:
Maybe we only need to remove the -opaque-pointers flag?

aschwaighofer commented:
The test was added here: https://github.com/apple/swift/pull/67964

yes, I don’t think we should remove this test—there doesn’t seem to be a copy elsewhere

bnbarham commented:
Ah okay, I was assuming `merge_func.ll` covered this. I'll remove the opaque pointer and rename the test to to something with casted return types then.

bnbarham committed:
[rebranch] Fix recently added merge function test

Opaque pointers are always enabled on rebranch. Rename the test to
better represent what it's testing.
